src/wolfsftp.c

```diff
@@ -1692,22 +1670,19 @@ int wolfSSH_SFTP_RecvMKDIR(WOLFSSH* ssh, int reqId, byte* data, word32 maxSz)
         return WS_BUFFER_E;
     }
 
-    /* plus one to make sure is null terminated */
-    dir = (char*)WMALLOC(sz + 1, ssh->ctx->heap, DYNTYPE_BUFFER);
-    if (dir == NULL) {
-        return WS_MEMORY_E;
+    ret = GetAndCleanPath(ssh->sftpDefaultPath,
+            data + idx, sz, dir, sizeof(dir));
+    if (ret != WS_SUCCESS) {
+        return WS_BUFFER_E;
```

Comment:
Why not just return ret, which has the actual error code that got thrown?

Reply:
Makes sense.